packages/react-native-gesture-handler/src/web/handlers/PanGestureHandler.ts

```diff
@@ -62,113 +49,136 @@ export default class PanGestureHandler extends GestureHandler {
+    super.updateGestureConfig(config);
 
-    if (this.config.minDist !== undefined) {
-      this.minDistSq = this.config.minDist * this.config.minDist;
-    } else if (this.hasCustomActivationCriteria) {
-      this.minDistSq = Number.MAX_SAFE_INTEGER;
+    if (config.minPointers !== undefined) {
+      this.minPointers = config.minPointers;
     }
 
-    if (this.config.minPointers !== undefined) {
-      this.minPointers = this.config.minPointers;
+    if (config.maxPointers !== undefined) {
+      this.maxPointers = config.maxPointers;
     }
 
-    if (this.config.maxPointers !== undefined) {
-      this.maxPointers = this.config.maxPointers;
+    if (config.minVelocity !== undefined) {
+      this.minVelocityX = config.minVelocity;
+      this.minVelocityY = config.minVelocity;
+      this.hasCustomActivationCriteria = true;
     }
 
-    if (this.config.minVelocity !== undefined) {
-      this.minVelocityX = this.config.minVelocity;
-      this.minVelocityY = this.config.minVelocity;
+    if (config.minVelocityX !== undefined) {
+      this.minVelocityX = config.minVelocityX;
+      this.hasCustomActivationCriteria = true;
     }
 
-    if (this.config.minVelocityX !== undefined) {
-      this.minVelocityX = this.config.minVelocityX;
+    if (config.minVelocityY !== undefined) {
+      this.minVelocityY = config.minVelocityY;
+      this.hasCustomActivationCriteria = true;
     }
 
-    if (this.config.minVelocityY !== undefined) {
-      this.minVelocityY = this.config.minVelocityY;
+    if (config.activateAfterLongPress !== undefined) {
+      this.activateAfterLongPress = config.activateAfterLongPress;
     }
 
-    if (this.config.activateAfterLongPress !== undefined) {
-      this.activateAfterLongPress = this.config.activateAfterLongPress;
+    if (config.activeOffsetXStart !== undefined) {
+      this.activeOffsetXStart = config.activeOffsetXStart;
+      this.hasCustomActivationCriteria = true;
     }
 
-    if (this.config.activeOffsetXStart !== undefined) {
-      this.activeOffsetXStart = this.config.activeOffsetXStart;
-
-      if (this.config.activeOffsetXEnd === undefined) {
-        this.activeOffsetXEnd = Number.MAX_SAFE_INTEGER;
-      }
+    if (config.activeOffsetXEnd !== undefined) {
+      this.activeOffsetXEnd = config.activeOffsetXEnd;
+      this.hasCustomActivationCriteria = true;
     }
 
-    if (this.config.activeOffsetXEnd !== undefined) {
-      this.activeOffsetXEnd = this.config.activeOffsetXEnd;
-
-      if (this.config.activeOffsetXStart === undefined) {
-        this.activeOffsetXStart = Number.MIN_SAFE_INTEGER;
-      }
+    if (
+      this.activeOffsetXStart !== undefined &&
+      this.activeOffsetXEnd === undefined
+    ) {
+      this.activeOffsetXEnd = Number.MAX_SAFE_INTEGER;
+    } else if (
+      this.activeOffsetXStart === undefined &&
+      this.activeOffsetXEnd !== undefined
+    ) {
+      this.activeOffsetXStart = Number.MIN_SAFE_INTEGER;
     }
```

Comment:
Is the logic with `Number.MAX_SAFE_INTEGER` needed? Wouldn't that be handled by `resetConfig`?